tools/build_langserver/langserver/analyzer.go

```diff
@@ -1,3 +1,199 @@
 package langserver
 
+import (
+	"core"
+	"parse"
+	"strings"
+
+	"parse/asp"
+	"tools/build_langserver/lsp"
+	"io/ioutil"
+	"fmt"
+)
+
 // TODO(bnmetrics): This file should contain functions to retrieve builtin and custom definitions of build defs
+
+// Analyzer is a wrapper around asp.Analyzer
+// This is being loaded into a handler on initialization
+type Analyzer struct {
+	parser *asp.Parser
+	BuiltIns map[string]*RuleDef
```

Comment:
I think this should probably be private - I can imagine us needing it to be threadsafe later on.